src/Forms/GridField/GridState_Data.php

```diff
@@ -77,14 +96,56 @@ public function __toString()
         return json_encode($this->toArray());
     }
 
+    /**
+     * Convert the state to an array including only value that differ from the default state.
+     * @return array
+     */
     public function toArray()
```

Comment:
We've changed toArray from being the current set of values to being the current set of values that differs from the defaults. A new method, `getDirtyArray()`, or an `$nonDefaultsOnly = false` argument is better?

Reply:
Splitting it into two different methods sounds sensible.